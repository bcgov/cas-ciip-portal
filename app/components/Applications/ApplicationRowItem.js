import React, {Component} from 'react';
import propTypes from 'prop-types';
import {Button, Badge} from 'react-bootstrap';

class ApplicationRowItem extends Component {
  static propTypes = {
    application: propTypes.shape({
      applicationId: propTypes.number,
      applicationStatus: propTypes.string,
      submissionDate: propTypes.string,
      facilityName: propTypes.string,
      operatorName: propTypes.string,
      reportingYear: propTypes.string,
      bcghgid: propTypes.string
    }).isRequired
  };

  render() {
    const {application} = this.props;
    console.log('my application', application);
    const statusBadgeColor = {
      attention: 'warning',
      pending: 'info',
      declined: 'danger',
      approved: 'success'
    };
    // Const url = `http://localhost:3000/public/dashboard/985719f1-7eae-4c49-88a9-7d6c8edc1ad4?application_id=${application.applicationId}`;
    // Const url = `https://metabase-wksv3k-dev.pathfinder.gov.bc.ca/public/dashboard/bb6a4b75-3a7f-4fab-9268-cb013ecfcb7b?application_id=${application.applicationId}`;
    const applicationDetails = `/application-details?application_id=${application.applicationId}&reportingyear=${application.reportingYear}&bcghgid=${application.bcghgid}`;

    return (
      <tr>
        <td>{application.applicationId}</td>
        <td>{application.operatorName}</td>
        <td>{application.facilityName}</td>
        <td>{application.submissionDate}</td>
        <td>
          <Badge
            pill
            style={{width: '100%'}}
            variant={statusBadgeColor[application.applicationStatus]}
          >
            {application.applicationStatus}
          </Badge>
        </td>
        <td>
          <Button href={applicationDetails} target="_blank" variant="primary">
            View Application
          </Button>
        </td>
      </tr>
    );
  }
}

export default ApplicationRowItem;
