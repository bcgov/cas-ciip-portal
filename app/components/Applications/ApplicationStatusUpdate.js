import React, {Component} from 'react';
import propTypes from 'prop-types';


class ApplicationStatusUpdate extends Component {
  static propTypes = {
    applicationStatus: propTypes.string.isRequired,
    setApplicationStatus: propTypes.func.isRequired
  };

  render() {
    const {applicationStatus, setApplicationStatus} = this.props;
    const statusBadgeColor = {
      pending: 'info',
      attention: 'warning',
      declined: 'danger',
      approved: 'success'
    };

    
}

export default ApplicationStatusUpdate;
