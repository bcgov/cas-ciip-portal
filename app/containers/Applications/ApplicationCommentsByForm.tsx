import React from 'react';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import moment from 'moment';

/*
 * The ApplicationComments renders all the comments on the various sections of the application
 */

interface Props {
  applicationReview;
  relay: RelayProp;
}

export const ApplicationCommentsByForm: React.FunctionComponent<Props> = props => {
  const {applicationReview} = props;
  const reviewComments = applicationReview
    ? applicationReview.reviewCommentsByApplicationReviewId.edges
    : null;

  return (
    <>
      <div key={applicationReview.id} className="review-box">
        <div>
          Set to <strong>{applicationReview.reviewStatus}</strong>
          &nbsp;
          <small>
            {moment(applicationReview.createdAt).format(
              'MMM Do YYYY, h:mm:ss a'
            )}
          </small>
        </div>
        <ul className="comment-box">
          {reviewComments.map(({node}) => (
            <li key={node.id} className="comments">
              <div>{node.description}</div>
              <div>
                <small>
                  {moment(node.createdAt).format('MMM Do YYYY, h:mm:ss a')}
                </small>
              </div>
            </li>
          ))}
        </ul>
      </div>

      <style jsx>{`
        .form-result-box .review-box {
          border-bottom: 1px solid #d2d2d2;
          margin-bottom: 20px;
        }
      `}</style>
    </>
  );
};

export default createFragmentContainer(ApplicationCommentsByForm, {
  applicationReview: graphql`
    fragment ApplicationCommentsByForm_applicationReview on ApplicationReview {
      id
      reviewStatus
      formResultId
      createdAt
      reviewCommentsByApplicationReviewId(first: 2147483647)
        @connection(
          key: "ApplicationCommentsByForm_reviewCommentsByApplicationReviewId"
        ) {
        edges {
          node {
            id
            description
            createdAt
          }
        }
      }
    }
  `
});
