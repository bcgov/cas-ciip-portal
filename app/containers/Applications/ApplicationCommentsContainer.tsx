import React, {useState} from 'react';
import {Collapse, Table, Button} from 'react-bootstrap';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import {ApplicationCommentsContainer_formResult} from 'ApplicationCommentsContainer_formResult.graphql';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faChevronDown, faChevronRight} from '@fortawesome/free-solid-svg-icons';
import ApplicationCommentsBox from './ApplicationCommentsByForm';

/*
 * The ApplicationComments renders all the comments on the various sections of the application
 */

interface Props {
  formResult: ApplicationCommentsContainer_formResult;
  relay: RelayProp;
  review: boolean;
}

export const ApplicationCommentsComponent: React.FunctionComponent<Props> = (
  props
) => {
  const {formResult, review} = props;
  const [isOpen, setIsOpen] = useState(false);
  const [showResolved, toggleShowResolved] = useState(false);
  return (
    <>
      <div key={formResult.id} className="form-result-box">
        <h3>
          {formResult.formJsonByFormId.name}{' '}
          <i
            aria-label="Collapse or expand comment"
            title="Collapse or expand comment"
            style={{float: 'right', marginRight: '10px'}}
            onClick={() => setIsOpen(!isOpen)}
          >
            {isOpen ? (
              <FontAwesomeIcon icon={faChevronRight} />
            ) : (
              <FontAwesomeIcon icon={faChevronDown} />
            )}
          </i>
        </h3>
        <Collapse in={!isOpen}>
          <div>
            <Table hover>
              <thead className="comments-header">
                <tr>
                  <th>General Comment</th>
                  {review ? (
                    <th style={{textAlign: 'right'}}>Resolve/Delete</th>
                  ) : null}
                </tr>
              </thead>
              <tbody>
                {formResult.internalGeneralComments.edges.map(({node}) => {
                  if (node.resolved && !showResolved) return null;
                  if (
                    !review &&
                    formResult.applicationByApplicationId
                      .applicationRevisionStatus.applicationRevisionStatus ===
                      'SUBMITTED'
                  )
                    return null;
                  return (
                    <ApplicationCommentsBox
                      key={node.id}
                      review={review}
                      reviewComment={node}
                      version={formResult.versionNumber}
                      formResultId={formResult.id}
                    />
                  );
                })}
              </tbody>
            </Table>
            <Table hover>
              <thead className="comments-header">
                <tr>
                  <th>Request Changes</th>
                  {review ? (
                    <th style={{textAlign: 'right'}}>Resolve/Delete</th>
                  ) : null}
                </tr>
              </thead>
              <tbody>
                {formResult.requestedChangeComments.edges.map(({node}) => {
                  if (node.resolved && !showResolved) return null;
                  if (
                    !review &&
                    formResult.applicationByApplicationId
                      .applicationRevisionStatus.applicationRevisionStatus ===
                      'SUBMITTED'
                  )
                    return null;
                  return (
                    <ApplicationCommentsBox
                      key={node.id}
                      review={review}
                      reviewComment={node}
                      version={formResult.versionNumber}
                      formResultId={formResult.id}
                    />
                  );
                })}
              </tbody>
            </Table>
            <Button
              variant="link"
              style={{textDecoration: 'none'}}
              onClick={() => toggleShowResolved(!showResolved)}
            >
              {showResolved ? 'Hide Resolved' : 'Show Resolved'}
            </Button>
          </div>
        </Collapse>
      </div>
      <style jsx global>{`
        .form-result-box {
          padding: 25px;
          margin-bottom: 20px;
          border: 1px solid #ff713a;
          border-radius: 5px;
          background-color: #fff4dcab;
          font-size: 14px;
        }

        .form-result-box h3 {
          border-bottom: 1px solid #888;
          padding-bottom: 10px;
          margin-bottom: 20px;
          font-size: 1.25rem;
        }

        .hide-title label.form-label {
          display: none;
        }
        i {
          cursor: pointer;
        }
        .comments-header {
          font-weight: bold;
          color: #036;
        }
        .comments-header th {
          border: 0;
        }
      `}</style>
    </>
  );
};

export default createFragmentContainer(ApplicationCommentsComponent, {
  formResult: graphql`
    fragment ApplicationCommentsContainer_formResult on FormResult {
      id
      versionNumber
      applicationByApplicationId {
        id
        rowId
        applicationRevisionStatus(versionNumberInput: $version) {
          applicationRevisionStatus
        }
      }
      formJsonByFormId {
        rowId
        name
      }
      internalGeneralComments(first: 2147483647)
        @connection(
          key: "ApplicationCommentsContainer_internalGeneralComments"
        ) {
        edges {
          node {
            id
            resolved
            ...ApplicationCommentsByForm_reviewComment
          }
        }
      }
      requestedChangeComments(first: 2147483647)
        @connection(
          key: "ApplicationCommentsContainer_requestedChangeComments"
        ) {
        edges {
          node {
            id
            resolved
            ...ApplicationCommentsByForm_reviewComment
          }
        }
      }
    }
  `
});
