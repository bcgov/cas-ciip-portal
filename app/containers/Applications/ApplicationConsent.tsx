import React from 'react';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import {useRouter} from 'next/router';
import {Button} from 'react-bootstrap';
import updateApplicationRevisionMutation from 'mutations/application/updateApplicationRevisionMutation';
import {ApplicationConsent_application} from 'ApplicationConsent_application.graphql';

interface Props {
  application: ApplicationConsent_application;
  relay: RelayProp;
}

export const ApplicationConsent: React.FunctionComponent<Props> = (props) => {
  const router = useRouter();

  const handleContinueClick = async () => {
    const {environment} = props.relay;

    const variables = {
      input: {
        id: props.application.latestDraftRevision.id,
        applicationRevisionPatch: {
          legalDisclaimerAccepted: true,
          versionNumber: props.application.latestDraftRevision.versionNumber
        }
      }
    };

    await updateApplicationRevisionMutation(environment, variables);

    router.push({
      pathname: '/reporter/application',
      query: {
        applicationId: props.application.id,
        version: props.application.latestDraftRevision.versionNumber
      }
    });
  };

  return (
    <Button variant="primary" size="lg" onClick={handleContinueClick}>
      Consent and continue
    </Button>
  );
};

export default createFragmentContainer(ApplicationConsent, {
  application: graphql`
    fragment ApplicationConsent_application on Application {
      id
      latestDraftRevision {
        id
        versionNumber
      }
    }
  `
});
