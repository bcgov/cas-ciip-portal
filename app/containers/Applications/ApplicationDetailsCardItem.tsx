import React, {useState} from 'react';
import {Button, Card, Collapse, Col, Row} from 'react-bootstrap';
import {createFragmentContainer, graphql} from 'react-relay';
import JsonSchemaForm, {AjvError, ErrorSchema} from '@rjsf/core';
import {FormJson} from 'next-env';
import {ApplicationDetailsCardItem_formResult} from '__generated__/ApplicationDetailsCardItem_formResult.graphql';
import {ApplicationDetailsCardItem_query} from '__generated__/ApplicationDetailsCardItem_query.graphql';
import customFields from 'components/Application/ApplicationDetailsCardItemCustomFields';
import SummaryFormArrayFieldTemplate from 'containers/Forms/SummaryFormArrayFieldTemplate';
import SummaryFormFieldTemplate from 'containers/Forms/SummaryFormFieldTemplate';
import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
import {customTransformErrors} from 'functions/customTransformErrors';
import useJsonSchemaDiff from 'hooks/useJsonSchemaDiff';

interface Props {
  // The form_result used by the fragment
  formResult: ApplicationDetailsCardItem_formResult;
  /** Optional prop (used in review-application only).
      An array of form results from the version that has been selected to be diffed from (older version) default: current version - 1 */
  diffFromResults?: any;
  /** Optional prop (used in review-application only).
      An array of form results from the version that has been selected to be diffed To (newer version) default: current version */
  diffToResults?: any;
  // The query prop used by the fragment
  query: ApplicationDetailsCardItem_query;
  // Boolean indicates whether or not to show the diff between selected versions
  showDiff: boolean;
  // Boolean indicates whether or not to liveValidate the results (true when rendered by the summary component)
  liveValidate: boolean;
  setHasErrors?: (...args: any[]) => void;
}

/*
 * The ApplicationDetails renders a summary of the data submitted in the application
 */
export const ApplicationDetailsCardItemComponent: React.FunctionComponent<Props> = ({
  formResult,
  diffFromResults,
  diffToResults,
  query,
  showDiff,
  liveValidate,
  setHasErrors
}) => {
  const {formJsonByFormId} = formResult;
  const {formJson} = formJsonByFormId;
  const {schema, uiSchema, customFormats} = formJson as FormJson;

  const formIdPrefix = `${formJsonByFormId.name
    .toLowerCase()
    .replace(' ', '-')}`;

  const transformErrors = (errors: AjvError[]) => {
    return customTransformErrors(errors, formJson);
  };

  const handleChange = (_formData, errorSchema: ErrorSchema) => {
    if (setHasErrors) {
      if (errorSchema?.__errors) {
        setHasErrors(true);
      } else {
        setHasErrors(false);
      }
    }
  };

  // Expands or collapses the form_result card
  const [isOpen, setIsOpen] = useState(false);

  // Select the correct form result to diff to by matching formJson slugs
  const diffTo = diffToResults?.find(
    ({node}) => node.formJsonByFormId.slug === formJsonByFormId.slug
  )?.node.formResult;

  // Select the correct form result to diff from by matching formJson slugs
  const diffFrom = diffFromResults?.find(
    ({node}) => node.formJsonByFormId.slug === formJsonByFormId.slug
  )?.node.formResult;

  const {formData, idDiffMap} = useJsonSchemaDiff(
    formResult,
    showDiff,
    formIdPrefix,
    diffFrom,
    diffTo
  );

  const classTag = formJsonByFormId.slug;
  // Override submit button for each form with an empty fragment
  // eslint-disable-next-line react/jsx-no-useless-fragment
  const buttonOverride = <></>;

  return (
    <Card
      style={{width: '100%', marginBottom: '10px'}}
      className={`${classTag} summary-card`}
    >
      <Card.Header className="summary-form-header">
        <Row>
          <Col md={6}>
            <h2>{formJsonByFormId.name}</h2>
          </Col>
          <Col md={1} style={{textAlign: 'right'}}>
            <Button
              aria-label="toggle-card-open"
              title="expand or collapse the card"
              variant="outline-dark"
              onClick={() => setIsOpen(!isOpen)}
            >
              {isOpen ? '+' : '-'}
            </Button>
          </Col>
        </Row>
      </Card.Header>
      <Collapse in={!isOpen}>
        <Card.Body>
          <JsonSchemaForm
            omitExtraData
            liveOmit
            liveValidate={liveValidate}
            ArrayFieldTemplate={SummaryFormArrayFieldTemplate}
            FieldTemplate={SummaryFormFieldTemplate}
            showErrorList={false}
            fields={customFields}
            customFormats={customFormats}
            transformErrors={transformErrors}
            schema={schema}
            idPrefix={formIdPrefix}
            uiSchema={uiSchema}
            onChange={handleChange}
            ObjectFieldTemplate={FormObjectFieldTemplate}
            formData={formData}
            formContext={{
              query,
              showDiff,
              idDiffMap
            }}
          >
            {buttonOverride}
          </JsonSchemaForm>
        </Card.Body>
      </Collapse>
      <style jsx global>{`
        h2 {
          font-size: 1.5rem;
          margin-bottom: 0.5rem;
          font-weight: 500;
          line-height: 1.2;
        }
        .diffFrom {
          background-color: rgba(243, 76, 96, 0.3);
        }
        .diffTo {
          background-color: rgba(70, 241, 118, 0.3);
        }
        .summary-form-header > .row {
          justify-content: space-between;
        }
        @media print {
          button,
          .btn {
            display: none !important;
          }
        }
      `}</style>
    </Card>
  );
};

export default createFragmentContainer(ApplicationDetailsCardItemComponent, {
  formResult: graphql`
    fragment ApplicationDetailsCardItem_formResult on FormResult {
      formResult
      formJsonByFormId {
        name
        slug
        formJson
      }
    }
  `,
  query: graphql`
    fragment ApplicationDetailsCardItem_query on Query {
      ...NaicsField_query
      ...ProductField_query
      ...ProductRowIdField_query
      ...FuelField_query
      ...FuelRowIdField_query
      ...EmissionCategoryRowIdField_query
    }
  `
});
