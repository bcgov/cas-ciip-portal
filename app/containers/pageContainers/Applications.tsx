import React, {Component} from 'react';
import {graphql} from 'react-relay';
import {ApplicationsQueryResponse} from '__generated__/ApplicationsQuery.graphql';
import DefaultLayout from '../../layouts/default-layout';
import ApplicationListContainer from '../Applications/ApplicationListContainer';

interface Props {
  query?: ApplicationsQueryResponse['query'];
}

class Applications extends Component<Props> {
  // In downstream component
  // onClick={(event) => handleApplicationsAction({action: Applications.SORT_ACTION})})
  // I didn't fully understand how to make this work, so I moved on with the way I did.
  static query = graphql`
    query ApplicationsQuery(
      $orderByField: String
      $direction: String
      $searchField: String
      $searchValue: String
    ) {
      query {
        session {
          ...Header_session
        }
        ...ApplicationListContainer_query
          @arguments(
            orderByField: $orderByField
            direction: $direction
            searchField: $searchField
            searchValue: $searchValue
          )
      }
    }
  `;

  state = {
    orderByField: 'operator_name',
    direction: 'ASC',
    orderByDisplay: 'Operator Name',
    searchField: null,
    searchValue: null,
    searchDisplay: 'No Filter'
  };

  static async getInitialProps() {
    return {
      variables: {
        orderByField: 'operator_name',
        direction: 'ASC',
        searchField: null,
        searchValue: null
      }
    };
  }

  toggleDirection = () => {
    this.state.direction === 'ASC'
      ? this.setState({direction: 'DESC'})
      : this.setState({direction: 'ASC'});
  };

  sortApplications = (event, eventKey) => {
    this.setState({
      orderByField: eventKey,
      orderByDisplay: event.target.text
    });
  };

  applySearchField = (event, eventKey) => {
    this.setState({
      searchField: eventKey,
      searchDisplay: event.target.text,
      searchValue: null
    });
  };

  applySearchValue = event => {
    if (this.state.searchField !== 'none') {
      this.setState({searchValue: event.nativeEvent.target[0].value});
    }
  };

  handleEvent = (event, eventKey) => {
    event.preventDefault();
    event.stopPropagation();
    event.persist();
    this[event.target.id](event, eventKey);
  };

  render() {
    const {query} = this.props;
    return (
      <>
        <DefaultLayout title="Applications" session={query.session}>
          <ApplicationListContainer
            query={query}
            orderByDisplay={this.state.orderByDisplay}
            searchDisplay={this.state.searchDisplay}
            direction={this.state.direction}
            orderByField={this.state.orderByField}
            searchField={this.state.searchField}
            searchValue={this.state.searchValue}
            handleEvent={this.handleEvent}
          />
        </DefaultLayout>
      </>
    );
  }
}

export default Applications;
