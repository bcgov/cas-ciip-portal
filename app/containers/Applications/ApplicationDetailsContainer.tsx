import React, {useEffect, useState} from 'react';
import {Dropdown, Form, Row, Col /*Button*/} from 'react-bootstrap';
import DropdownMenuItemComponent from 'components/DropdownMenuItemComponent';
import {createRefetchContainer, graphql, RelayRefetchProp} from 'react-relay';
import {ApplicationDetailsContainer_query} from 'ApplicationDetailsContainer_query.graphql';
import {ApplicationDetailsContainer_application} from 'ApplicationDetailsContainer_application.graphql';
import ApplicationDetailsCardItem from './ApplicationDetailsCardItem';
// import FileDownload from 'js-file-download';

/*
 * The ApplicationDetails renders a summary of the data submitted in the application,
 * and allows the user to submit their application.
 */

interface Props {
  query: ApplicationDetailsContainer_query;
  application: ApplicationDetailsContainer_application;
  relay: RelayRefetchProp;
  review: boolean;
  // Boolean indicates whether or not this item is being rendered by the summary component & should be liveValidated
  liveValidate: boolean;
  setHasErrors?: (...args: any[]) => void;
  setApplicationDetailsRendered?: (boolean) => void;
}

export const ApplicationDetailsComponent: React.FunctionComponent<Props> = (
  props
) => {
  const formResults = props.application.orderedFormResults.edges;
  const diffFromResults = props.review
    ? props?.query?.old?.orderedFormResults?.edges
    : undefined;

  const [oldDiffVersion, setOldDiffVersion] = useState(
    (
      props.application.orderedFormResults.edges[0].node.versionNumber - 1
    ).toString()
  );
  const [newDiffVersion, setNewDiffVersion] = useState(
    props.application.orderedFormResults.edges[0].node.versionNumber.toString()
  );
  const [showDiff, setShowDiff] = useState(false);

  useEffect(() => {
    const refetchVariables = {
      oldVersion: oldDiffVersion,
      newVersion: newDiffVersion,
      applicationId: props.application.id
    };
    props.relay.refetch(refetchVariables);
  }, [oldDiffVersion, newDiffVersion, props.application.id, props.relay]);

  // const handleDownloadPdf = async () => {
  //   const response = await fetch('/print-pdf', {
  //     method: 'POST',
  //     headers: {
  //       'Content-Type': 'application/json'
  //     },
  //     body: JSON.stringify({html: document.documentElement.outerHTML})
  //   });
  //   const blob = await response.blob();
  //   FileDownload(blob, 'CIIP_Application.pdf');
  // };

  if (props.setApplicationDetailsRendered)
    props.setApplicationDetailsRendered(true);
  return (
    <>
      <Row>
        <Col md={{offset: 2, span: 2}}>
          {diffFromResults ? (
            <Form.Check
              label="Show Diff?"
              checked={showDiff}
              type="checkbox"
              onChange={() => setShowDiff(!showDiff)}
            />
          ) : null}
        </Col>
        {showDiff ? (
          <>
            <Col md={2}>
              <Dropdown style={{width: '100%', textTransform: 'capitalize'}}>
                <Dropdown.Toggle
                  style={{width: '100%', textTransform: 'capitalize'}}
                  id="dropdown-old"
                >
                  {oldDiffVersion === '0'
                    ? 'swrs import'
                    : 'Version '.concat(oldDiffVersion)}
                </Dropdown.Toggle>
                <Dropdown.Menu style={{width: '100%'}}>
                  {props.application.applicationRevisionsByApplicationId.edges.map(
                    ({node}, index) =>
                      index >= Number(newDiffVersion) ||
                      index === Number(oldDiffVersion) ? null : (
                        <DropdownMenuItemComponent
                          key={node.id}
                          itemEventKey={node.versionNumber}
                          itemFunc={setOldDiffVersion}
                          itemTitle={
                            node.versionNumber === 0
                              ? 'swrs import'
                              : node.versionNumber
                          }
                        />
                      )
                  )}
                </Dropdown.Menu>
              </Dropdown>
            </Col>
            <Col md={1}>&nbsp;------&gt;</Col>
            <Col md={2}>
              <Dropdown style={{width: '100%', textTransform: 'capitalize'}}>
                <Dropdown.Toggle
                  style={{width: '100%', textTransform: 'capitalize'}}
                  id="dropdown-new"
                >
                  {Number(newDiffVersion) ===
                  props.application.latestSubmittedRevision.versionNumber
                    ? `current (V${newDiffVersion})`
                    : 'Version '.concat(newDiffVersion)}
                </Dropdown.Toggle>
                <Dropdown.Menu style={{width: '100%'}}>
                  {props.application.applicationRevisionsByApplicationId.edges.map(
                    ({node}, index) =>
                      index <= Number(oldDiffVersion) ? null : (
                        <DropdownMenuItemComponent
                          key={node.id}
                          itemEventKey={node.versionNumber}
                          itemFunc={setNewDiffVersion}
                          itemTitle={
                            node.versionNumber ===
                            props.application.latestSubmittedRevision
                              .versionNumber
                              ? `current (V${node.versionNumber})`
                              : node.versionNumber
                          }
                        />
                      )
                  )}
                </Dropdown.Menu>
              </Dropdown>
            </Col>
          </>
        ) : null}
      </Row>
      <br />

      <div>
        {formResults.map(({node}) => (
          <ApplicationDetailsCardItem
            key={node.id}
            liveValidate={props.liveValidate}
            diffFromResults={diffFromResults}
            diffToResults={props.query.new.orderedFormResults.edges}
            formResult={node}
            query={props.query.query}
            review={props.review}
            showDiff={showDiff}
            setHasErrors={props.setHasErrors}
          />
        ))}
      </div>
      {/* <Button variant="primary" onClick={handleDownloadPdf}>
        Download PDF
      </Button> */}
      <style jsx global>{`
        @media print {
          header .header-right,
          #navbar,
          footer {
            display: none !important;
          }
        }
      `}</style>
    </>
  );
};

export default createRefetchContainer(
  ApplicationDetailsComponent,
  {
    query: graphql`
      fragment ApplicationDetailsContainer_query on Query
      @argumentDefinitions(
        applicationId: {type: "ID!"}
        newVersion: {type: "String"}
        oldVersion: {type: "String"}
      ) {
        query {
          ...ApplicationDetailsCardItem_query
        }
        old: application(id: $applicationId) {
          orderedFormResults(versionNumberInput: $oldVersion) {
            edges {
              node {
                id
                versionNumber
                formJsonByFormId {
                  slug
                }
                formResult
              }
            }
          }
        }
        new: application(id: $applicationId) {
          orderedFormResults(versionNumberInput: $newVersion) {
            edges {
              node {
                id
                versionNumber
                formJsonByFormId {
                  slug
                }
                formResult
              }
            }
          }
        }
      }
    `,
    application: graphql`
      fragment ApplicationDetailsContainer_application on Application
      @argumentDefinitions(version: {type: "String!"}) {
        id
        orderedFormResults(versionNumberInput: $version) {
          edges {
            node {
              id
              versionNumber
              ...ApplicationDetailsCardItem_formResult
            }
          }
        }
        latestSubmittedRevision {
          versionNumber
        }
        applicationRevisionsByApplicationId {
          totalCount
          edges {
            node {
              id
              versionNumber
            }
          }
        }
      }
    `
  },
  graphql`
    query ApplicationDetailsContainerRefetchQuery(
      $oldVersion: String
      $newVersion: String
      $applicationId: ID!
    ) {
      query {
        ...ApplicationDetailsContainer_query
          @arguments(
            oldVersion: $oldVersion
            newVersion: $newVersion
            applicationId: $applicationId
          )
      }
    }
  `
);
