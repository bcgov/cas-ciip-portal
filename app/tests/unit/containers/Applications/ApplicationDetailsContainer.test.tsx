import React from 'react';
import {shallow} from 'enzyme';
import {ApplicationDetailsComponent} from 'containers/Applications/ApplicationDetailsContainer';

describe('ApplicationDetailsComponent', () => {
  it('should match the snapshot with the ApplicationDetails component', async () => {
    const renderer = shallow(
      <ApplicationDetailsComponent
        isAnalyst
        relay={null}
        query={{
          ' $refType': 'ApplicationDetailsContainer_query',
          query: null
        }}
        application={{
          ' $refType': 'ApplicationDetailsContainer_application',
          id: 'abc',
          formResultsByApplicationId: {
            edges: [
              {
                node: {
                  id: 'WyJmb3JtX3Jlc3VsdHMiLDExXQ==',
                  ' $fragmentRefs': {
                    ApplicationDetailsCardItem_formResult: true
                  }
                }
              }
            ]
          }
        }}
      />
    );
    expect(renderer).toMatchSnapshot();
  });
});
