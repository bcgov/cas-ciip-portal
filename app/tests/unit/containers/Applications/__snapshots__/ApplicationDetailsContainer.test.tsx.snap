// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDetailsComponent should match the snapshot with the ApplicationDetails component 1`] = `
<Fragment>
  <Row
    noGutters={false}
  >
    <Col
      md={
        Object {
          "offset": 2,
        }
      }
    />
  </Row>
  <br
    className="jsx-2472284510"
  />
  <div
    className="jsx-2472284510"
  >
    <Relay(ApplicationDetailsCardItemComponent)
      diffToResults={
        Array [
          Object {
            "node": Object {
              "formJsonByFormId": Object {
                "slug": "admin",
              },
              "formResult": null,
              "id": "abc",
              "versionNumber": 2,
            },
          },
        ]
      }
      formResult={
        Object {
          " $fragmentRefs": Object {
            "ApplicationDetailsCardItem_formResult": true,
          },
          "id": "WyJmb3JtX3Jlc3VsdHMiLDExXQ==",
          "versionNumber": 2,
        }
      }
      key="WyJmb3JtX3Jlc3VsdHMiLDExXQ=="
      liveValidate={false}
      query={
        Object {
          " $fragmentRefs": Object {
            "ApplicationDetailsCardItem_query": true,
          },
        }
      }
      review={false}
      showDiff={false}
    />
  </div>
  <JSXStyle
    id="2472284510"
  >
    @media print{header .header-right,#navbar,footer{display:none !important;}}
  </JSXStyle>
</Fragment>
`;
